lib/TopStoriesFeed.jsm

```diff
@@ -169,8 +166,10 @@ this.TopStoriesFeed = class TopStoriesFeed {
       this.updateSettings(data.stories.settings);
       this.stories = this.rotate(this.transform(stories));
       this.storiesLastUpdated = data.stories._timestamp;
-      if (spocs && spocs.length) {
-        this.spocs = spocs;
+      if (data.stories.spocs && data.stories.spocs.length) {
```

Comment:
So instead of looking at post processed spocs that we cached, we look at the un processed spocs cached in body (which gets renamed to stories). Then just do the same processing to them that the fetch does.